src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectlyTests.cs

```diff
@@ -0,0 +1,658 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Usage.CSharpImplementGenericMathInterfacesCorrectly,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Usage.UnitTests
+{
+    public class ImplementGenericMathInterfacesCorrectlyTests
+    {
+        [Fact]
+        public async Task IParsableNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System;
+
+public readonly struct MyDate : IParsable<{|#0:DateOnly|}> // The 'IParsable<TSelf>' interface requires the 'TSelf' type parameter to be filled with the derived type 'MyDate' 
```

Comment:
Can we have a test like:

```csharp
using IParsableOfDateOnly = IParsable<DateOnly>;

public readonly struct MyDate : IParsableOfDateOnly
{
    // ....
}
```

The syntactic checks in the analyzer could fail? I think we shouldn't rely on syntax at all.

Reply:
Good scenario, thanks, I will add a test

> The syntactic checks in the analyzer could fail?  I think we shouldn't rely on syntax at all.

There will be no type parameter found yes, but I don't think we should stop trying to locate type parameter because of this, it is just a case where we will squiggle on the symbol, probably we need a bit different message for this